packages/dds/tree/src/core/tree/visitDelta.ts

```diff
@@ -408,39 +411,51 @@ function detachPass(delta: Delta.FieldChanges, visitor: DeltaVisitor, config: Pa
 				visitNode(index, mark.fields, visitor, config);
 			}
 			if (isDetachMark(mark)) {
-				for (let i = 0; i < mark.count; i += 1) {
-					const root = config.detachedFieldIndex.createEntry(
-						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-						offsetDetachId(mark.detach!, i),
-					);
-					if (mark.fields !== undefined) {
-						config.attachPassRoots.set(root, mark.fields);
-					}
-					const field = config.detachedFieldIndex.toFieldKey(root);
-					visitor.detach({ start: index, end: index + 1 }, field);
+				/* Create a single root for a contiguous range of detached nodes  */
+				const root = config.detachedFieldIndex.createEntry(mark.detach, mark.count);
+				if (mark.fields !== undefined) {
+					config.attachPassRoots.set(root, mark.fields);
 				}
+				const field = config.detachedFieldIndex.toFieldKey(root);
+				visitor.detach({ start: index, end: index + mark.count }, field);
 			} else if (!isAttachMark(mark)) {
 				index += mark.count;
 			}
 		}
 	}
 }
 
+/**
+ * For the non-atomization purpose of the tree delta visit behavior, we will partition the existing
+ * detached nodes into multiple segments. This partitioning is determined by whether the nodes within
+ * a segment share the same root or not. If the shared root is undefined, the tree-building process
```

Comment:
I'm not sure what "share the same root or not" means here, but those are likely not the right words. It's also not clear why how this comment relates to this function specifically, so perhaps that comment should be somewhere else.

Reply:
Comments updated

Reply:
This comment is still wrong an confusing/confused. Saying nodes "have an indentical root" means that they share some ancestor in a tree topology. This is not at all what this function is dealing with. You probably meant "have an identical ForestRootId", but that is also wrong: there is a 1:1 mapping between the nodes and the `ForestRootID`s. So when you see some count of nodes `N` with the first node having node ID `X`, there is some `ForestRootID` `Y` and that corresponds to node `X`, but also some `ForestRootID` `Y+1` that corresponds to node `X+1` etc, for all N nodes.

I don't think we need a comment explaining the implementation details of the function (the callers don't need to know, and it's the sort of comment that is liable to not be updated when the implementation changes), so you can just remove it.